src/suites/cts/robust_access.spec.ts

```diff
@@ -0,0 +1,364 @@
+export const description = `Tests to check array clamping in shaders is correctly implemented including vector / matrix indexing`;
+
+import { TestGroup, pcombine, pfilter, poptions } from '../../framework/index.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export const g = new TestGroup(GPUTest);
+
+// Utilities that should probably live in some shared place.
+function copyArrayBuffer(src: ArrayBuffer): ArrayBuffer {
+  const dst = new ArrayBuffer(src.byteLength);
+  new Uint8Array(dst).set(new Uint8Array(src));
+  return dst;
+}
+
+const kUintMax = 4294967295;
+const kIntMax = 2147483647;
+
+// A small utility to test shaders:
+//  - it wraps the source into a small harness that checks the runTest() function returns 0.
+//  - it runs the shader with the testBindings set as bindgroup 0.
+//
+// The shader also has access to a uniform value that's equal to 0u to avoid constant propagation
+// in the shader compiler.
+async function runShaderTest(
```

Comment:
```suggestion
function runShaderTest(
```
don't think async is needed